### DISCUSSION TITLE
A Post Interaction Settings option to set to only the posters followers can reply to a post?

### DISCUSSION THREAD

**Discussion by ScoredOne** [bluesky-social/atproto#2896] [Bluesky Client App]
Hello, something I noticed from the art community that came over from past couple days. Some larger artists were wanting to have posts where current followers of theirs could post their art in the the replys of a post, only for it to be picked up by the wider communities and random people were posting their art in there too.

The current options are: Everybody, Nobody, Mentioned Users and Followed Users.

I'm not familiar at all at the moment with Blueskys structure as I have only taken peaks myself and with my current situation but if we are tracking the follow date of a user to the poster, it would be feasible to add an option to only allow for users that were following before the posts created date to reply to the post? Obviously something like this would need checking against the overhead of performing such a check on the database ext, hence why I'm posting here.

**Comment by surfdude29** [comment]
This is an intriguing idea. I'm just speculating here, and I should emphasise that I don't know much about how the threadgate works, but I imagine perhaps the simplest way of implementing a "Followers at the time of this post" option would be to simply generate a list of DIDs who follow you at the time the post is created with that threadgate option set.

Then the threadgate would just check whether a user is on that static list to know if they're allowed to reply.

A possibly big downside to this approach though would be that a separate, static list of users would need to be stored in the user's repo on the PDS for each post where this threadgate option is set. And if someone used this option often, it might add a lot of extra data to the user's repo.

There may well be an easier and better way to implement this, I just thought I'd add my musings in case they might be helpful :)
